main.tf

```diff
@@ -1,3 +1,213 @@
+        rate_increase_percent        = var.auto_scaling.disk.rate_increase_percent
+        rate_limit_mb_per_member     = var.auto_scaling.disk.rate_limit_mb_per_member
+        rate_period_seconds          = var.auto_scaling.disk.rate_period_seconds
+        rate_units                   = var.auto_scaling.disk.rate_units
+      }
+      memory {
+        io_above_percent         = var.auto_scaling.memory.io_above_percent
+        io_enabled               = var.auto_scaling.memory.io_enabled
+        io_over_period           = var.auto_scaling.memory.io_over_period
+        rate_increase_percent    = var.auto_scaling.memory.rate_increase_percent
+        rate_limit_mb_per_member = var.auto_scaling.memory.rate_limit_mb_per_member
+        rate_period_seconds      = var.auto_scaling.memory.rate_period_seconds
+        rate_units               = var.auto_scaling.memory.rate_units
+      }
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      # Ignore changes to these because a change will destroy and recreate the instance
+      version,
+      key_protect_key,
+      backup_encryption_key_crn
+    ]
+  }
+
+  timeouts {
+    create = "120m" # Extending provisioning time to 120 minutes
+  }
+}
+
+##############################################################################
+# Context Based Restrictions
+##############################################################################
+
+module "cbr_rule" {
+  count            = length(var.cbr_rules) > 0 ? length(var.cbr_rules) : 0
+  source           = "terraform-ibm-modules/cbr/ibm//cbr-rule-module"
+  version          = "1.2.0"
+  rule_description = var.cbr_rules[count.index].description
+  enforcement_mode = var.cbr_rules[count.index].enforcement_mode
+  rule_contexts    = var.cbr_rules[count.index].rule_contexts
+  resources = [{
+    attributes = [
+      {
+        name     = "accountId"
+        value    = var.cbr_rules[count.index].account_id
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceInstance"
+        value    = ibm_database.redis_database.id
+        operator = "stringEquals"
+      },
+      {
+        name     = "serviceName"
+        value    = "databases-for-redis"
+        operator = "stringEquals"
+      }
+    ]
+  }]
+  #  There is only 1 operation type for Redis so it is not exposed as a configuration
+  operations = [{
+    api_types = [
+      {
+        api_type_id = "crn:v1:bluemix:public:context-based-restrictions::::api-type:data-plane"
+      }
+    ]
+  }]
+}
+
+##############################################################################
+# Service Credentials
+##############################################################################
+
+resource "ibm_resource_key" "service_credentials" {
+  for_each             = var.service_credential_names
+  name                 = each.key
+  role                 = each.value
+  resource_instance_id = ibm_database.redis_database.id
+  tags                 = var.tags
```

Comment:
This is a bug that was spotted yesterday across ICD modules. Resource key tagging is not actually supported by IBM Cloud. The provider doc actually says this:
![image](https://github.com/terraform-ibm-modules/terraform-ibm-icd-redis/assets/29608224/82592059-9d6d-4f53-8809-df43f48992c0)

So I think remove the tags attribute from this block here, and also update the variable description for `tags` so it doesn't mention service credentials.